rotkehlchen/db/accounting_rules.py

```diff
@@ -248,8 +256,58 @@ def missing_accounting_rules(self, events: list[HistoryBaseEntry]) -> list[bool]
                         ),
                     )
                     missing_accounting_rule.append(cursor.fetchone()[0] == 0)
-                else:
-                    missing_accounting_rule.append(is_missing_rule)
+                    continue
+
+                # check the surrounding events to verify if this particular event will
+                # be processed by a previous one. First query the related events
+                related_events = history_db.get_history_events(
+                    cursor=cursor,
+                    filter_query=HistoryEventFilterQuery.make(event_identifiers=[event.event_identifier]),
+                    has_premium=True,
+                )
+                for related_event_idx, related_event in enumerate(related_events):
+                    counterparty = getattr(event, 'counterparty', None)
+                    if counterparty == CPT_GAS:  # avoid checking the case of gas in evm events
+                        continue
+
+                    # query the accounting rule for the related event
+                    cursor.execute(
+                        'SELECT accounting_treatment FROM accounting_rules WHERE (type=? AND subtype=? AND (counterparty=? OR counterparty=?))',  # noqa: E501
+                        (
+                            related_event.event_type.serialize(),
+                            related_event.event_subtype.serialize(),
+                            NO_ACCOUNTING_COUNTERPARTY if counterparty is None else counterparty,
+                            NO_ACCOUNTING_COUNTERPARTY,
+                        ),
+                    )
+                    if (raw_treatment := cursor.fetchone()) is None:
+                        continue
+
+                    accounting_treatment = TxAccountingTreatment.deserialize_from_db(raw_treatment[0])  # noqa: E501
+                    match accounting_treatment:
+                        case TxAccountingTreatment.SWAP:
+                            # ensure that the event that we are going to check can exist
+                            # in the related events. Avoid index error
+                            if len(related_events) <= related_event_idx + 1:
```

Comment:
1. Don't use yoda conditionals. They are too confusing
2. If something is used multiple times in a for loop define **once** outside of it.
```suggestion
                            if related_event_idx + 1 > related_events_length:
```

But why do you need this condition? The comment seems misleading. if you hit this continue it's over, right?